src/Builder/ResponseBuilder.php

```diff
@@ -135,8 +135,8 @@ public function addHeader($headerLine)
                 sprintf('"%s" is not a valid HTTP header line', $headerLine)
             );
         }
-        $name = trim(urldecode($parts[0]));
-        $value = trim(urldecode($parts[1]));
+        $name = trim($parts[0]);
+        $value = trim($parts[1]);
```

Comment:
the urldecode was in there from the beginning. is something else doing that decoding in reality? is your test failing if you keep the urldecode here?